drivers/misc/coresight/nrf_etr.c

```diff
@@ -84,6 +443,43 @@ static void rd_idx_inc(void)
 	etr_rd_idx += CORESIGHT_TRACE_FRAME_SIZE32;
 }
 
+static void decode(uint8_t *buf, uint32_t pending)
```

Comment:
It looks like this function not only decodes the structure but also outputs it using UART. It should be renamed for clarity, or reimplemented to match the name.

Reimplementation seems cleaner because it would allow adding more outputs than UART easier.